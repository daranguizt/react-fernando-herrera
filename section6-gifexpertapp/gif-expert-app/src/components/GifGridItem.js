import React, { Fragment } from "react";
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__bounce animate__fadeIn">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  );
};

export default GifGridItem;

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}