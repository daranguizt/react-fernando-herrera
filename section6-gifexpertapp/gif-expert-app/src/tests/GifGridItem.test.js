import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../components/GifGridItem";

describe("gifgriditem tests", () => {
  const testObj = {
    id: 1,
    title: "test",
    url: "https://media.giphy.com/media/d1E2IByItLUuONMc/giphy.gif",
    key: "https://media.giphy.com/media/d1E2IByItLUuONMc/giphy.gif",
  };

  const wrapper = shallow(
    <GifGridItem
      id={testObj.id}
      title={testObj.title}
      url={testObj.url}
      key={testObj.key}
    />
  );
  test("should show gifgriditem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have a paragraph with title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(testObj.title);
  });

  test('should have img with given url', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(testObj.url);
  });

  test('should have animate__fadeIn', () => {
    const div = wrapper.find('.animate__fadeIn');
    expect(div.length).toBe(1);

    //soluci√≥n fernando
    //const div = wrapper.find('div');
    //expect(div.props('className').includes('animate__fadeIn')).toBe(true);
  })
});
