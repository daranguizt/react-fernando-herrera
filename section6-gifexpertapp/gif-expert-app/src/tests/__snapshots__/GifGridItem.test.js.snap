// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gifgriditem tests should show gifgriditem component 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="test"
    src="https://media.giphy.com/media/d1E2IByItLUuONMc/giphy.gif"
  />
  <p>
    test
  </p>
</div>
`;
